### PR TITLE
#395: Created a C++ file and updated readme for Shortest Common Super-sequence

### PR DESCRIPTION
**Pull request by AastikSharma05**
## Related Issue
- Given two strings str1 and str2, the task is to find the length of the shortest string that has both str1 and str2 as subsequences. (please mention issue number #395  )
- Added the code implementation of Shortest Common Super-sequence Problem using dp.
- Added the explanation of Shortest Common Super-sequence Problem in the README.md file.

## Proposed Changes
- Added the code of Shortest Common Super-sequence Problem using dp.
- Added README.md file of Shortest Common Super-sequence Problem with time and space complexity.

## Additional Info
- This code uses dynamic programming.
- This problem is a variation of LCS problem.
- This is one of the commonly asked interview questions and the concept of this question helps us to solve many more problems.
- This code solves an open issue #395 
- I have tested these changes on VS code using multiple inputs and matched the output with the expected output.
- This change will not affect or create conflict with any other code
- Time Complexity : O(m*n)
- Space Complexity: O(m*n)

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

Issue: #395 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| Readme preview  | Code Snippet     |
| Screenshot 2022-03-08 134618 | Screenshot 2022-03-06 140830 |

### CODE REVIEW

**Comment by AastikSharma05** [comment]
please review my pr @iamgunjan @miraj0507

**Comment by Lakhankumawat** [comment]
Resolve the conflicts manually

**Comment by AastikSharma05** [comment]
I have resolved the conflict so please review my pr and tell if anything needs to be done @siddhi-244 @Lakhankumawat

**Comment by AastikSharma05** [comment]
@siddhi-244 @miraj0507 @Lakhankumawat plz review it

**Comment by miraj0507** [comment]
Hey, kindly take an user input in the main function

**Comment by AastikSharma05** [comment]
@miraj0507 I have made the changes

**Comment by miraj0507** [comment]
LGTM

**Comment by AastikSharma05** [comment]
If everything is fine then please merge this pr @Lakhankumawat

**Review by n4i9kita — COMMENTED** [review]
LGTM

File: D-DynamicProgramming/README.md [hunk 1]
```diff
@@ -166,15 +160,74 @@ Time Complexity: O(sum*n)
...
 - [Problem Statement](#problem-statement)
     - [Examples](#examples)
 - [Explanation](#explanation)
 - [Complexity](#complexity)
 
 # Problem Statement
 
+
+Given two strings s1 and s2, you have to output the shortest length of the string that has both strings s1 and s2 as subsequences.
+
```
Comment (Lakhankumawat) [inline D-DynamicProgramming/README.md:174]: Add a working image 


You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

File: D-DynamicProgramming/Shortest_Common_Super-sequence.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
+// C++ program to find length of the
```
Comment (Lakhankumawat) [inline D-DynamicProgramming/Shortest_Common_Super-sequence.cpp:1]: Rename code file to `ShortestCommonSuperSequence.cpp`

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

**Review by AastikSharma05 — COMMENTED** [review]
I have made all the changes.

**Comment by AastikSharma05** [comment]
@Lakhankumawat Please check now.

**Review by Lakhankumawat — APPROVED** [review]
LGTM 🎉✨

**Comment by Lakhankumawat** [comment]
@all-contributors please add @AastikSharma05 code

**Comment by Lakhankumawat** [comment]
Looking at your contribution type , contribution  efforts and 2 merged prs  i am increasing the level of PR 

CGC= Continuous Good Contribution

**Comment by AastikSharma05** [comment]
Thanks a lot @Lakhankumawat .

**Pull request merged**